### PR TITLE
check lost/not reported tests fail build

### PR DESCRIPTION
**Pull request by sergioboi**

### CODE REVIEW

File: subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/InHouseInstrumentationTestRunner.kt [hunk 1]
```diff
@@ -171,6 +172,7 @@ abstract class InHouseInstrumentationTestRunner :
      */
     @SuppressLint("LogNotTimber")
     override fun onCreate(arguments: Bundle) {
+        throw RuntimeException("test")
```
Comment (eugene-krivobokov) [inline subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/InHouseInstrumentationTestRunner.kt]: We can make a feature toggle to test this case :smiley:

**Comment by sergioboi** [comment]
Problem was because we had exception in `com.avito.runner.service.worker.device.adb.AdbDevice#runTest` chain. It leads us to skip report sending in `doOnSuccess`. Then we got empty report and decide there weren't failed test.
Error is tricky cause we send lost after test run and make report not empty.

File: subprojects/gradle/runner/service/src/main/kotlin/com/avito/runner/service/model/intention/Action.kt [hunk 2]
```diff
@@ -3,19 +3,16 @@ package com.avito.runner.service.model.intention
...
+    val targetPackage: String,
+    val testRunner: String,
+    val instrumentationParams: Map<String, String>,
+    val executionNumber: Int,
+    val timeoutMinutes: Long
+) {
+    override fun toString(): String = "Run ${test.testName} test"
 }
 
 sealed class ActionResult {
```
Comment (eugene-krivobokov) [inline subprojects/gradle/runner/service/src/main/kotlin/com/avito/runner/service/model/intention/Action.kt]: Do we still need this sealed?

File: subprojects/gradle/runner/client/src/test/kotlin/com/avito/runner/scheduler/runner/RunnerIntegrationTest.kt [hunk 3]
```diff
@@ -849,7 +850,7 @@ class RunnerIntegrationTest {
 
     private fun provideRunner(
         observer: DevicesObserver,
-        testListener: TestListener? = null,
+        testListener: TestListener = EmptyListener,
```
Comment (eugene-krivobokov) [inline subprojects/gradle/runner/client/src/test/kotlin/com/avito/runner/scheduler/runner/RunnerIntegrationTest.kt]: :+1:

File: subprojects/gradle/runner/shared-test/src/main/kotlin/com/avito/runner/test/EmptyListener.kt [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package com.avito.runner.test
+
+import com.avito.runner.service.listener.TestListener
+import com.avito.runner.service.model.TestCase
+import com.avito.runner.service.model.TestCaseRun
+import com.avito.runner.service.worker.device.Device
+
+object EmptyListener: TestListener {
```
Comment (eugene-krivobokov) [inline subprojects/gradle/runner/shared-test/src/main/kotlin/com/avito/runner/test/EmptyListener.kt]: `Empty` -> `NoOp` ?

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/report/HasFailedTestDeterminer.kt [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+interface HasFailedTestDeterminer {
+
+    fun determine(
+        runResult: Try<List<SimpleRunTest>>
+    ): Result
+
+    sealed class Result {
+
+        open fun count(): Int = 0
```
Comment (eugene-krivobokov) [inline subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/report/HasFailedTestDeterminer.kt:14]: Default implementation can break LSP

File: subprojects/gradle/runner/service/src/test/kotlin/com/avito/runner/service/worker/device/adb/AdbDeviceTest.kt [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package com.avito.runner.service.worker.device.adb
+
+internal class AdbDeviceTest {
```
Comment (eugene-krivobokov) [inline subprojects/gradle/runner/service/src/test/kotlin/com/avito/runner/service/worker/device/adb/AdbDeviceTest.kt:3]: why is it empty?

File: subprojects/gradle/runner/shared/src/main/kotlin/com/avito/runner/exit/ExitManager.kt [hunk 7]
```diff
@@ -1,14 +0,0 @@
-package com.avito.runner.exit
-
-import org.gradle.api.GradleException
-
-interface ExitManager {
-    fun exit(exit: Exit)
-}
-
-class GradleExitManager : ExitManager {
```
Comment (eugene-krivobokov) [inline subprojects/gradle/runner/shared/src/main/kotlin/com/avito/runner/exit/ExitManager.kt:9]: :+1:

**Review by eugene-krivobokov — APPROVED** [review]

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/report/HasFailedTestDeterminer.kt (hunk 5 above)
Comment (sergioboi) [reply subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/report/HasFailedTestDeterminer.kt:14]: `Функции, которые используют базовый тип, должны иметь возможность использовать подтипы базового типа, не зная об этом.`
How does it break LSP?

Comment (eugene-krivobokov) [reply subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/report/HasFailedTestDeterminer.kt:14]: I agree. It's maybe not about LSP.

DetermineError(error).count() -> 0, this confused me, that count() makes no sense in some children

**Pull request merged**